### DISCUSSION TITLE
Allow expressions for slice values

### DISCUSSION THREAD

**Discussion by innovate-invent** [jmespath-community/jmespath.spec#67] [Ideas]
With `let()` and unquoted literals we can have dynamic slice expressions.

Replace
```
slice-expression  = [number] ":" [number] [ ":" [number] ]
```
with
```
slice-expression  = [expression / number ] ":" [expression / number ] [ ":" [expression / number ] ]
```

**Comment by springcomp** [comment]
Can you elaborate on one scenario where this would be useful?

I think you suggest using expressions that consist in using the `let()` function inside `slice-expressions`. However, the `let()` function does not have a defined return value so I fail to see how this would behave?

**Reply by innovate-invent** [reply]
A trivial example would be cropping two arrays to the same length:
```
search({short: [ a, b, c ], long: [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ]}, long[:length( short )] )
```
